gstcefsrc.cc

```diff
@@ -709,6 +723,11 @@ gst_cef_src_class_init (GstCefSrcClass * klass)
           "Enable GPU usage in chromium (Improves performance if you have GPU)",
           DEFAULT_GPU, (GParamFlags) (G_PARAM_READWRITE | G_PARAM_STATIC_STRINGS | GST_PARAM_MUTABLE_READY)));
 
+  g_object_class_install_property (gobject_class, PROP_CHROMIUM_DEBUG_PORT,
+    g_param_spec_int ("chromium-debug-port", "chromium-debug-port",
+          "Set chromium debug port (0 = disabled)", 0, G_MAXUINT16,
+          DEFAULT_CHROMIUM_DEBUG_PORT, (GParamFlags) (G_PARAM_READWRITE | G_PARAM_STATIC_STRINGS | GST_PARAM_MUTABLE_READY)));
```

Comment:
For some reason default value always shows up as `0` when I do gst-inspect. I wanted to go with `-1`, but it didn't work :( 
Checked that someone else had had a similar experience, but didn't find a reason why (http://gstreamer-devel.966125.n4.nabble.com/why-I-can-t-use-g-param-spec-int-set-default-value-td4660762.html)
So went with 0

Reply:
You have to set the field to the default value in the `_init()` function.

Reply:
Hmm not sure I understand. Need a bit of spoonfeeding :D . Could you point to the place?
From what I see `url` prop is done in similar way and it works

Reply:
https://developer.gnome.org/gobject/stable/howto-gobject-construction.html:

> /* initialize all public and private members to reasonable default values.
>  * They are all automatically initialized to 0 to begin with. */

This works with URL because 0 is NULL :)

Besides that, gst-inspect by default will get_property to determine the default, there is an optional flag you could set for it to show the "actual" default, but this isn't relevant here.

Shouldn't we only allow >= 1024 though? In that case 0 would be a fair default anyway

Reply:
Could change the range, but then what would be the disabled port value? :thinking:

Reply:
hah, good point good point, `G_MAXUINT16 + 1` :P

Reply:
but yeah no, -1 -> G_MAXUINT16, -1 disabled sounds fine

Reply:
But now how can I put the default value as -1? :D If I just change `DEFAULT_CHROMIUM_DEBUG_PORT` to -1 it doesnt work :( 
I checked https://developer.gnome.org/gobject/stable/howto-gobject-construction.html and it just puts `2` as default value there. Or am I missing something? :D 
If I put value in `gst_cef_src_init` then prop is not shoving at all with gst-inspect

Reply:
you need to set the minimum as -1 when registering the property

Reply:
it's really not that complicated, maybe go read the doc for install_property, and remember that gst-inspect-1.0 will *not* show the default you provide there, but instead g_object_get() the value after istantiating the element (for reasons, which you shouldn't worry about :P)

Reply:
Ok hmm I added `src->chromium_debug_port = DEFAULT_CHROMIUM_DEBUG_PORT;` as well and now it seems to work. 
So `This works with URL because 0 is NULL :)` how come? I'm so confused about this design decision. Why my `-1` doesn't get picked up but a string does? :thinking:

Reply:
And seems like chromium is ignoring 0 port value by default.

Reply:
Because the default initialization of an object (`g_object_new` effectively) sets everything to 0/NULL.  If you need something else in the object, you need to explicitly set that value inside `_init()`.

Reply:
as @ystreet said, @reinismu I feel for you if you're not used to writing (gobject) C, but if that comforts you there's no magic involved :)

Changes look OK now, can you fix up your commits, rebase on master and push again? I will merge afterwards :)

Reply:
To me, magic feels that how the default URL is set without being inside `_init()` part :D